components/concept/engine/src/main/java/mozilla/components/concept/engine/search/SearchRequest.kt

```diff
@@ -0,0 +1,15 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.search
+
+/**
+ * Value type that represents a request for showing a search to the user.
+ */
+data class SearchRequest(val type: Type, val query: String) {
+    /**
+     * Whether the search should be made in a private or normal session.
+     */
+    enum class Type { NORMAL, PRIVATE }
```

Comment:
We've largely just used a boolean to represent private mode in Android Components. Could that be used instead?

Reply:
I prefer enums here, but I'm happy to follow the local convention.  Changing now.